static/js/notifications.js

```diff
@@ -259,6 +259,10 @@ function process_notification(notification) {
     }
 
     if (message.type === "private") {
+        if (page_params.hide_private_message_desktop_notifications !== undefined
+            && page_params.hide_private_message_desktop_notifications) {
+            content = "You got a message from " + message.sender_full_name;
+        }
```

Comment:
Can you post a screenshot of what this looks like?  You can test using two browsers logged into different users (or incognito windows, same effect)